examples/plot_simulate_evoked_newapi.py

```diff
@@ -0,0 +1,150 @@
+"""
+===============
+Simulate dipole using new drives-API
+===============
+
+This example demonstrates how to simulate a dipole for evoked-like
+waveforms using HNN-core.
+
+Note that the output will be slightly different from HNN-GUI due to different
+"seeds" being used when creating the random exogeneous input spikes to the
+network. The results should match qualitatively, however.
+"""
+
+# Authors: Mainak Jas <mainak.jas@telecom-paristech.fr>
+#          Sam Neymotin <samnemo@gmail.com>
+#          Blake Caldwell <blake_caldwell@brown.edu>
+#          Christopher Bailey <cjb@cfin.au.dk>
+
+import os.path as op
+import tempfile
+
+###############################################################################
+# Let us import hnn_core
+
+import hnn_core
+from hnn_core import simulate_dipole, read_params, Network, read_spikes
+from hnn_core.viz import plot_dipole
+
+hnn_core_root = op.dirname(hnn_core.__file__)
+
+###############################################################################
+# Then we read the parameters file
+params_fname = op.join(hnn_core_root, 'param', 'default.json')
+params = read_params(params_fname)
+print(params)
+
+###############################################################################
+# This is a lot of parameters! We can also filter the
+# parameters using unix-style wildcard characters
+print(params['L2Pyr_soma*'])
+
+###############################################################################
+# Let us first create our network from the params file and visualize the cells
+# inside it. The default behaviour of Network is to add and instantiate six
+# 'default' drives, but we will suppress that by setting the
+# `initialise_hnn_drives`-argument to `False`.
+net = Network(params, initialise_hnn_drives=False)
+net.plot_cells()
+
+###############################################################################
+# The network of cells is now defined, to which we add external drives as
+# required. Weights are prescribed separately for AMPA and NMDA receptors
+# (receptors that are not used can be omitted or set to zero)
+
+# Distal evoked drive
+weights_ampa = {'L2_basket': 0.006562, 'L2_pyramidal': .000007,
+                'L5_pyramidal': 0.142300}
+weights_nmda = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
+                'L5_pyramidal': 0.080074}
+dispersion_time = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                   'L5_pyramidal': 0.1}
+net.add_evoked_drive(
+    'evdist1', mu=63.53, sigma=3.85, numspikes=1, weights_ampa=weights_ampa,
+    weights_nmda=weights_nmda, location='distal', seedcore=2,
+    space_constant=3., dispersion_time=dispersion_time)
+
+# First proximal evoked drive
+weights_ampa = {'L2_basket': 0.08831, 'L2_pyramidal': 0.01525,
+                'L5_basket': 0.19934, 'L5_pyramidal': 0.00865}
+weights_nmda = {'L2_basket': 0., 'L2_pyramidal': 0.,
+                'L5_basket': 0., 'L5_pyramidal': 0.}  # all weights zero
```

Comment:
what do you think of making a shortcut `weights_nmda = 0.` like for the dispersion parameter?

Reply:
actually, `weights_nmda = 0.` won't work... `weights_nmda = {}` does, and for any `cellname` that's omitted, the effect is to set that cell type weight to zero.

Allowing passing a float for weights seems dodgy to me: is there a conceivable use case in which all cell types had the same weights?

Reply:
>  that's omitted, the effect is to set that cell type weight to zero.

okay excellent. Perhaps then do `weights_nmda = {}` up here so it's documented? I know many folks would want to turn off specific cell-type connectivity.

Reply:
> is there a conceivable use case in which all cell types had the same weights?

not that I can think of. Let's say YAGNI for now

Reply:
Agree that the float passing seems weird, maybe a `None` shortcut?

Reply:
None should be used for the default ...